src/cdktf/cdktf-construct.ts

```diff
@@ -1,12 +1,20 @@
-import { ConstructLibrary, ConstructLibraryOptions } from '../cdk';
+import { ConstructLibrary, ConstructLibraryOptions } from "../cdk";
+import * as semver from "semver";
 
 export interface ConstructLibraryCdktfOptions extends ConstructLibraryOptions {
   /**
    * Minimum target version this library is tested against.
-   * @default "0.4.0"
+   * @default "0.8.3"
    * @featured
    */
   readonly cdktfVersion: string;
+
+  /**
+   * Construct version to use
+   * @default "10.0.12"
+   * @featured
+   */
+  readonly constructsVersion?: string;
```

Comment:
This is actually the minimum version. Let's change this to be semver requirement so users can express a richer set:

```suggestion
  /**
   * Construct version to use
   * @default "^10.0.12"
   */
  readonly constructsVersion?: string;
```

Also, we try to reduce the set of "featured" options to a minimum. This is unlikely something someone would really want to touch.